lib/features/snack_queue/controllers/default_snack_controller.dart

```diff
@@ -0,0 +1,78 @@
+// ignore_for_file: no-equal-arguments
+import 'package:flutter/material.dart';
+import 'package:flutter_easy_dialogs/flutter_easy_dialogs.dart';
+import 'package:flutter_template/features/snack_queue/domain/entities/snack_message_type.dart';
+import 'package:flutter_template/uikit/app_sizes.dart';
+import 'package:flutter_template/uikit/colors/app_color_scheme.dart';
+import 'package:flutter_template/uikit/text/app_text_scheme.dart';
+
+/// Default controller for displaying messages.
+class DefaultSnackController {
+  /// A context is needed to access the theme.
+  final BuildContext context;
+
+  /// Create an instance [DefaultSnackController].
+  const DefaultSnackController.from(this.context);
```

Comment:
I think it is not a good idea to hold `BuildContext` as a field in the class. Can we add `BuildContext` as a parameter to the `showSnack` method?